### DISCUSSION TITLE
Trouble compiling Advent4 Any Ideas?

### DISCUSSION THREAD

**Discussion by Technikfreak2002** [picocomputer/community#29] [Q&A] [answered]
Hello!

I just finished my picocomputer build! Thank you for this awsome project!

I also set up VSCode and the scaffolding.

I successfuly compiled an uploaded "Hello World" and even ehBasic works just fine!

But if i want to build advent4, i got a wired error during linking. 

Here the output of VSCode:

`
[build] [100%] Linking C executable advent4
[build] common/_mappederrno.s:25: Error: Unresolved external '___osmaperrno'
[build] ld65: Error: 1 unresolved external(s) found - cannot create output file
[build] gmake[3]: *** [CMakeFiles/advent4.dir/build.make:367: advent4] Fehler 1
[build] gmake[2]: *** [CMakeFiles/Makefile2:100: CMakeFiles/advent4.dir/all] Fehler 2
[build] gmake[1]: *** [CMakeFiles/Makefile2:107: CMakeFiles/advent4.dir/rule] Fehler 2
[build] gmake: *** [Makefile:124: advent4] Fehler 2
[proc] Der Befehl "/home/markus/.local/bin/cmake --build /home/markus/pico/PiPicoComputer/advent4/build --config Debug --target advent4 -j 26 --" wurde mit dem Code "2" beendet.
[driver] Build abgeschlossen: 00:00:00.118
[build] Der Build wurde mit dem Exitcode 2 abgeschlossen.
`

I can not find anything about this error:
`common/_mappederrno.s:25: Error: Unresolved external '___osmaperrno`

I am working on an UpToDate LinuxMint 21.2 All tools a fresh installed

I hope someone could help me a bit.

Thanks!

Electrix6502

**Comment by brentward** [comment]
I don't know what the exact cause of this is but it looks like there is a discrepancy between the version of cc65 you are using and what the sdk is expecting.  If I look in the cc65 repo in github at the asminc/errno.inc file (https://github.com/cc65/cc65/blob/ce3bcadd24d3af05dc30f542722f6efb66382e82/asminc/errno.inc#L4) I can see the symbol matches what your error says, that is with 3 underscores (___osmaperrno). But if  you look in the sdk (https://github.com/picocomputer/rp6502-sdk/blob/634c60c3b981a1a083a05b38e11868ae7ab5154e/src/oserror.s) the symbol actually only has 2 underscores (__osmaperrno). In my local copy of cc65 on a mac the file /usr/local/Cellar/cc65/2.19/share/cc65/asminc/errno.inc matches the sdk with only 2 underscores (__osmaperrno).

I'm not positive that this is the cause but maybe you are using a newer version of cc65? I've only tested building on a mac using the cc65 version installed by homebrew and on ubuntu (in WSL2) using the version installed with apt (under 22.04).  On my Mac this is the version response
```
$ cc65 -V
cc65 V2.18 - N/A
```

**Comment by Technikfreak2002** [comment]
Hello! Thanks for the fast reply!

cc65 -V
cc65 V2.19 - Git ce3bcad

is a minor version newer. but normally installed via apt-get

**Reply by brentward** [reply]
You have the latest commit in master. The change in question was done Aug 28, 2022. This seems likely the reason it isn't compiling. This is the commit in question:
https://github.com/cc65/cc65/commit/5d390489a8f754abfd032af858b6ed5686553187

I have the release version of 2.19 (which reports in error as 2.18), you have the 2.19 with the latest Git commit (commit ce3bcad). You should try installing an older version.
Try this:

First uninstall cc65
```
sudo apt remove cc65
```
then find available versions
```
apt list --all-versions cc65
```
then install the appropriate version, preferably the one that is 2.19 but not Git.
```
sudo apt install cc65=<package version>
```

**Reply by Technikfreak2002** [reply]
Thank you very much!
I tried the downgrade inside a VM and yes, that was the issue. Seems i have fetched the source only a few hours before the update :D

**Comment by rumbledethumps** [comment, marked as answer]
It hadn't been updated to the new SDK which has a fix for some recent CC65 changes.

**Reply by rumbledethumps** [reply]
I thought I typed "update the SDK" somewhere but it seems I didn't. The latest SDK exports both old and new osmaperrno symbols.
https://github.com/picocomputer/rp6502-sdk/blob/e7b940052cf8c581815be8a3fb006dc14b81a76c/src/oserror.s#L16-L17

**Comment by Technikfreak2002** [comment]
Thank you very much!
With the updated SDK it compiles without errors!
